src/python_testing/TC_JFADMIN_2_2.py

```diff
@@ -0,0 +1,251 @@
+            "sdk-config": {
+                "ExampleOpCredsCAKey1": jfcStorage.get("Default", "ExampleOpCredsCAKey0"),
+                "ExampleOpCredsICAKey1": jfcStorage.get("Default", "ExampleOpCredsICAKey0"),
+                "ExampleCARootCert1": jfcStorage.get("Default", "ExampleCARootCert0"),
+                "ExampleCAIntermediateCert1": jfcStorage.get("Default", "ExampleCAIntermediateCert0"),
+            },
+            "repl-config": {
+                "caList": {
+                    "1": [
+                        {
+                            "fabricId": 1,
+                            "vendorId": self.jfctrl_fabric_a_vid
+                        }
+                    ]
+                }
+            }
+        }
+        # Extract CATs to be provided to the Python Controller later
+        self.ecoACATs = base64.b64decode(jfcStorage.get("Default", "CommissionerCATs"))[::-1].hex().strip('0')
+
+    def teardown_class(self):
+        # Stop all Subprocesses that were started in this test case
+        if self.fabric_a_admin is not None:
+            self.fabric_a_admin.terminate()
+        if self.fabric_a_ctrl is not None:
+            self.fabric_a_ctrl.terminate()
+
+        super().teardown_class()
+
+    def steps_TC_JFADMIN_2_2(self) -> list[TestStep]:
+        return [
+            TestStep("2", "TH sends ICACCSRRequest command to DUT.",
+                     "DUT response contains status code FAILSAFE_REQUIRED."),
+            TestStep("3", "TH sends ArmFailSafe command to DUT with ExpiryLengthSeconds set to 10 and Breadcrumb 1.",
+                     "DUT respond with ArmFailSafeResponse Command."),
+            TestStep("4", "TH sends ICACCSRRequest command to DUT.",
+                     "DUT response contains an ICACCSR."),
+            TestStep("5", "Wait for ArmFailSafe to expire."),
+            TestStep("6", "TH sends AddICAC command to DUT using icac1 as parameter.",
+                     "DUT response contains status code FAILSAFE_REQUIRED."),
+            TestStep("7", "TH sends ArmFailSafe command to DUT with ExpiryLengthSeconds set to 10 and Breadcrumb 1.",
+                     "DUT respond with ArmFailSafeResponse Command."),
+            # TestStep("8", "[SKIP Missing SDK Implementation] TH sends AddICAC command to DUT using icac1 as parameter.",
+            #          "DUT ICACResponse contains status 2 (InvalidICAC).")
+        ]
+
+    @async_test_body
+    async def test_TC_JFADMIN_2_2(self):
+
+        # Creating a Controller for Ecosystem A
+        _fabric_a_persistent_storage = PersistentStorage(jsonData=self.ecoACtrlStorage)
+        _certAuthorityManagerA = CertificateAuthority.CertificateAuthorityManager(
+            chipStack=self.matter_stack._chip_stack,
+            persistentStorage=_fabric_a_persistent_storage)
+        _certAuthorityManagerA.LoadAuthoritiesFromStorage()
+        devCtrlEcoA = _certAuthorityManagerA.activeCaList[0].adminList[0].NewController(
```

Comment:
Need to call Shutdown on devCtrlEcoA when no longer needed. Please apply this fix to all JF TCs.